source/extensions/filters/network/thrift_proxy/router/router_impl.cc

```diff
@@ -503,6 +508,9 @@ void Router::UpstreamRequest::onPoolReady(Tcp::ConnectionPool::ConnectionDataPtr
     upgrade_response_ =
         protocol_->attemptUpgrade(*transport_, *conn_state_, parent_.upstream_request_buffer_);
     if (upgrade_response_ != nullptr) {
+      parent_.recordClusterScopeHistogram({parent_.upstream_rq_size_},
```

Comment:
I can't recall if these upgrade requests and responses are tracked as request and responses. It may warrant a note in the stats doc, to clarify any confusion about the counters and histograms not matching up in the number of measurements.

Reply:
They are not accounted as independent reqs/resp -- so I think it's probably fine to treat them as part of their corresponding req/resp and accumulate & increment accordingly. What do you think?

Reply:
I went with this and added a line about it in the docs.